app/api/header/route.ts

```diff
@@ -0,0 +1,40 @@
+import { NextRequest, NextResponse } from 'next/server';
+import { type Locale } from 'next-intl';
+import { ZodError } from 'zod';
+
+import { errors, loggerErrors } from '~/constants/errors';
+
+import { createRequestContainer } from '~/di/container';
+import logger from '~/middleware/logger/logger';
+import { createHeaderService } from '~/services/composed/header-service/headerService';
+import { ResponseError } from '~/shared/exceptions/errors/responseError';
+
+export async function GET(req: NextRequest) {
+  const locale = req.nextUrl.searchParams.get('locale') || 'uk';
+
+  try {
+    const container = createRequestContainer();
+    const navigationService = container.resolve('navigationService');
+    const foundationInfoService = container.resolve('foundationInfoService');
+
+    const headerService = createHeaderService({ navigationService, foundationInfoService });
```

Comment:
```suggestion
    const container = createRequestContainer();
    const headerService = container.resolve('headerService');
```
You can resolve `headerService` with its cores services inside with this one line of code, i tried it and its work
If we use a DI-container, then we need to follow the rules of layered architecture and not import the service directly